### PR TITLE
8315869: UseHeavyMonitors not used

### PR DESCRIPTION
**Pull request by coleenp**
Please review this trivial change to remove the UseHeavyMonitors develop option, in favor of the now non-experimental LockingMode=LM_MONITOR (0) option.  Tested with tier1 locally.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315869](https://bugs.openjdk.org/browse/JDK-8315869): UseHeavyMonitors not used (**Enhancement** - P4)

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15845/head:pull/15845` \
`$ git checkout pull/15845`

Update a local copy of the PR: \
`$ git checkout pull/15845` \
`$ git pull https://git.openjdk.org/jdk.git pull/15845/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15845`

View PR using the GUI difftool: \
`$ git pr show -t 15845`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15845.diff">https://git.openjdk.org/jdk/pull/15845.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15845#issuecomment-1728179135)

### CODE REVIEW

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -1059,13 +1059,9 @@ const int ObjectAlignmentInBytes = 8;
...
           "If true, error data is printed to stdout instead of a file")     \
                                                                             \
-  develop(bool, UseHeavyMonitors, false,                                    \
-          "(Deprecated) Use heavyweight instead of lightweight Java "       \
-          "monitors")                                                       \
-                                                                            \
   develop(bool, VerifyHeavyMonitors, false,                                 \
           "Checks that no stack locking happens when using "                \
-          "+UseHeavyMonitors")                                              \
+          "-XX:LockingMode=LM_MONITOR")                                     \
```
Comment (dcubed-ojdk) [inline src/hotspot/share/runtime/globals.hpp:1064]: s/LM_MONITOR/0/

I don't think you can specify the `LM_MONITOR` value on the actual cmd line.

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up.

**Comment by coleenp** [comment]
Thanks Dan.  Do you agree that it's trivial?

**Review by AlanBateman — APPROVED** [review]

**Comment by dcubed-ojdk** [comment]
Yes, I agree that this is a trivial fix. (Sorry I missed that earlier.)

**Comment by coleenp** [comment]
Thanks Dan and Alan for the reviews.
/integrate

**Pull request closed without merging**